internal/io/file/file_sink_test.go

```diff
@@ -439,15 +634,209 @@ func TestFileSinkRollingCount_Collect(t *testing.T) {
 			// Should write to 2 files
 			for i := 0; i < 3; i++ {
 				// Read the contents of the temporary file and check if they match the collected items
-				fn := fmt.Sprintf("test_%s_%d.dd", tt.ft, 460+450*i)
+				var fn string
+				if tt.compress != "" {
+					fn = fmt.Sprintf("test_%s_%s_%d.dd", tt.ft, tt.compress, 460+450*i)
+				} else {
+					fn = fmt.Sprintf("test_%s_%d.dd", tt.ft, 460+450*i)
+				}
+
 				contents, err := os.ReadFile(fn)
 				if err != nil {
 					t.Fatal(err)
 				}
+				if tt.compress != "" && tt.compress != NONE_COMPRESS {
+					decompressor, _ := compressor.GetDecompressor(tt.compress)
+					contents, err = decompressor.Decompress(contents)
+					if err != nil {
+						t.Errorf("%v", err)
+					}
+				}
 				if !reflect.DeepEqual(contents, tt.contents[i]) {
 					t.Errorf("\nexpected\t %q \nbut got\t\t %q", tt.contents[i], string(contents))
 				}
 			}
 		})
 	}
 }
+
+func TestFileSinkCompress_Collect(t *testing.T) {
```

Comment:
It seems `TestFileSinkCompress_Collect` both test the compress for fileSink and fileSource, can you split the testcase into 2 parts and move the `fileSource` testcase into `file_source_test`?